web/src/stores/clips.ts

```diff
@@ -1,16 +1,71 @@
 import { Action as ReduxAction, Dispatch } from 'redux';
-const contributableLocales = require('../../../locales/contributable.json') as string[];
+const contributableLocales =
+  require('../../../locales/contributable.json') as string[];
 import StateTree from './tree';
 import { User } from './user';
 import { Clip } from 'common';
 
+async function getCanPlayAudio(audioSrc: string) {
+  return new Promise(resolve => {
+    const audio = new Audio(audioSrc);
+
+    audio.addEventListener('error', () => {
+      resolve(false);
+    });
+
+    audio.addEventListener('canplay', () => {
+      resolve(true);
+    });
+
+    audio.load();
+  });
+}
+
+async function checkClipsForErrors(clips: Clip[]) {
+  return await Promise.all(
+    clips.map(async clip => {
+      let hasError = false;
+
+      try {
+        // can play clip
```

Comment:
Rather than doing a `fetch` and downloading the whole file, we can use the Audio API to detect as soon as they are ready to 'play', this should be a bit faster and also gives us the benefit of checking whether they're valid audio files too

Reply:
This is great, thanks so much!